CompanyApi/Controllers/CompanyController.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Runtime.InteropServices.ComTypes;
+using CompanyApi.Model;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CompanyApi.Controllers
+{
+    [ApiController]
+    [Route("companies")]
+    public class CompanyController : ControllerBase
+    {
+        private static List<Company> companies = new List<Company>();
+
+        [HttpPost]
+        public ActionResult<Company> AddNewCompany(Company company)
+        {
+            if (companies.Any(existCompany => existCompany.Name == company.Name))
+            {
+                return Conflict();
+            }
+
+            company.CompanyId = Guid.NewGuid().ToString();
+
+            companies.Add(company);
+            return new CreatedResult($"/companies/{company.CompanyId}", company);
+        }
+
+        [HttpDelete]
+        public void DeleteAllCompanies()
+        {
+            companies.Clear();
+        }
+
+        [HttpGet]
+        public ActionResult<List<Company>> GetAllCompanies([FromQuery] int? pageSize, [FromQuery] int? pageIndex)
+        {
+            if (pageSize != null && pageIndex != null)
+            {
+                // return companies.GetRange((pageIndex.Value - 1) * pageSize.Value, pageSize.Value);
+                return companies
```

Comment:
should return ok(companies........),
 and no logic of situation if any query is null or both are null